lib/view_component/base.rb

```diff
@@ -99,13 +99,12 @@ def render?
 
     def initialize(*); end
 
-    # If trying to render a partial or template inside a component,
-    # pass the render call to the parent view_context.
+    # use original view_context if we're not rendering a component
     def render(options = {}, args = {}, &block)
-      if options.is_a?(String) || (options.is_a?(Hash) && options.has_key?(:partial))
-        view_context.render(options, args, &block)
-      else
+      if options.is_a? self.class
```

Comment:
```suggestion
      if options.is_a? ViewComponent::Base
```

I think we want `ViewComponent::Base` instead of `self.class`, otherwise I think we'll hit the `view_context.render` case unless we're rendering the same component recursively.